src/odemis/driver/xt_client.py

```diff
@@ -1469,7 +1498,19 @@ def __init__(self, name, role, parent, channel_name, **kwargs):
         self._shape = (256,)  # Depth of the image
         self.data = SEMDataFlow(self)
 
-        self._channel_name = channel_name
+        if hasattr(self.parent, "_scanner") and hasattr(self.parent, "_fib_scanner"):
+            self.active_scanner_type = StringEnumerated("electron", choices={"electron", "ion"},
+                                                        setter=self._set_active_scanner_type)
+        elif hasattr(self.parent, "_scanner"):
+            self._channel_name = channel_name
+            self._active_scanner = self.parent._scanner
+        elif hasattr(self.parent, "_fib_scanner"):
+            self._channel_name = channel_name
+            self._active_scanner = self.parent._fib_scanner
+        else:
+            raise ValueError("Non of the required scanner children are provided. Provide at least an ebeam scanner ("
+                             "single of multi beam) or a FIB scanner.")
```

Comment:
The parent already checks for that normally. So you can keep the message  simple, like "No scanner available".